### DISCUSSION TITLE
tail calls

### DISCUSSION THREAD

**Discussion by unicomp21** [WebAssembly/binaryen#4731] [Q&A]
@tlively, the tail call support in binaryen, does that mean I can do co_wait stuff now?  I also noticed in the blazor.net stuff they also managed to implement the "tail" instruction somehow for webassembly?

**Comment by tlively** [comment]
See previous discussion at https://github.com/emscripten-core/emscripten/issues/10991. LLVM and Binaryen support tail calls, but I don't believe they have been shipped in any Web engine yet, so you can't actually use them in practice. It's possible co_wait will just work but be inefficient or sometimes exhaust the stack, but I haven't tried.
